include/aws/mqtt/client.h

```diff
@@ -110,6 +110,40 @@ struct aws_mqtt_topic_subscription {
     void *on_publish_ud;
 };
 
+/**
+ * host_name                 The server name to connect to. This resource may be freed immediately on return.
+ * port                      The port on the server to connect to
+ * client_id                 The clientid to place in the CONNECT packet.
+ * socket_options            The socket options to pass to the aws_client_bootstrap functions.
+ *                           This is copied into the connection
+ * tls_options               TLS settings to use when opening a connection.
+ *                           This is copied into the connection
+ *                           Pass NULL to connect without TLS (NOT RECOMMENDED)
+ * clean_session             True to discard all server session data and start fresh
+ * keep_alive_time           The keep alive value to place in the CONNECT PACKET, a PING will automatically
+ *                           be sent at this interval as well. If you specify 0, defaults will be used
+ *                           and a ping will be sent every 60s. If you're tempted to set this really low
+ *                           to find out about connection drops quickly, this is going to consume a lot of cpu.
+ *                           Instead, consider setting tcp keep alive to low timeout values and leaving the ping
+ *                           at the max value allowed by the broker.
+ * request_timeout_ms        Amount of time before a PING is considered timed out. Default is 3s.
+ *                           Specify 0 for defaults.
+ * on_connection_complete    The callback to fire when the connection attempt completes
+ * user_data                  Passed to the userdata param of on_connection_complete
+ */
+struct aws_mqtt_connection_options {
+    const struct aws_byte_cursor *host_name;
```

Comment:
I have a lot of options structs/parameters that use aws_string pointers (with non-owner-transfer semantics).  Would you advise me to follow suit and use byte cursors for those?

Reply:
I dunno, this isn't really a change from the existing code, it just moved stuff around.

these are always copied, so they should be cursors.